tst/EnergyPlus/unit/WCE_Reader.unit.cc

```diff
@@ -0,0 +1,185 @@
+                          "0.7,                     !- Front Side Slat Beam Visible Reflectance",
+                          "0.7,                     !- Back Side Slat Beam Visible Reflectance",
+                          "0,                       !- Slat Diffuse Visible Transmittance",
+                          "0.7,                     !- Front Side Slat Diffuse Visible Reflectance",
+                          "0.7,                     !- Back Side Slat Diffuse Visible Reflectance",
+                          "0,                       !- Slat Infrared Hemispherical Transmittance",
+                          "0.9,                     !- Front Side Slat Infrared Hemispherical Emissivity",
+                          "0.9,                     !- Back Side Slat Infrared Hemispherical Emissivity",
+                          "0.0127,                  !- Blind to Glass Distance {m}",
+                          "0.0,                     !- Blind Top Opening Multiplier",
+                          "0.0,                     !- Blind Bottom Opening Multiplier",
+                          "0.0,                     !- Blind Left Side Opening Multiplier",
+                          "0.0,                     !- Blind Right Side Opening Multiplier",
+                          "0,                       !- Minimum Slat Angle {deg}",
+                          "0;                       !- Maximum Slat Angle {deg}"
+                          });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    HeatBalanceManager::GetMaterialData(ErrorsFound);
+    HeatBalanceManager::GetConstructData(ErrorsFound);
+    WindowManager::initWindowModel();
+    WindowManager::InitWindowOpticalCalculations();
+    HeatBalanceManager::InitHeatBalance();
+
+    auto aWinConstSimp = WindowManager::CWindowConstructionsSimplified::instance();
+    auto solarLayer = aWinConstSimp.getEquivalentLayer(FenestrationCommon::WavelengthRange::Solar, 1);
+
+    // Transmittance Front
+    const auto Tfront = solarLayer->getPropertySimple(FenestrationCommon::PropertySimple::T, FenestrationCommon::Side::Front, 
+        FenestrationCommon::Scattering::DirectDirect);
+    EXPECT_NEAR(0.8239, Tfront, 1e-6);
+
+    // Reflectance Front
+    const auto Rfront = solarLayer->getPropertySimple(FenestrationCommon::PropertySimple::R, FenestrationCommon::Side::Front,
+        FenestrationCommon::Scattering::DirectDirect);
+    EXPECT_NEAR(0.073578, Rfront, 1e-6);
+
+    // Transmittance Back
+    const auto Tback = solarLayer->getPropertySimple(FenestrationCommon::PropertySimple::T, FenestrationCommon::Side::Back,
+        FenestrationCommon::Scattering::DirectDirect);
+    EXPECT_NEAR(0.8239, Tback, 1e-6);
+
+    // Reflectance Back
+    const auto Rback = solarLayer->getPropertySimple(FenestrationCommon::PropertySimple::R, FenestrationCommon::Side::Back,
+        FenestrationCommon::Scattering::DirectDirect);
+    EXPECT_NEAR(0.073682, Rback, 1e-6);
+}
```

Comment:
Succinct, which I don't mind, but is it sufficiently covering the E+ implementation?  It seems like you have plenty of coverage of WCE itself in there, but I am more concerned about the E+ mechanics and making sure we hit as many corner cases as possible.

Reply:
This part should call not just reading but also thermal and optical part of WCE calculation within EnergyPlus. Yes, I am pulling numbers out, but these numbers are obtained by calling EnergyPlus routines. Will check this again on Monday.